proposal/forward-with.md

```diff
@@ -166,6 +162,10 @@ Given a source file `file`, a configuration `config`, and an import context
   * Let `module` be the result of [loading][] the module with `rule`'s URL
     and `rule-config`.
 
+  * If `rule` has a `WithClause` that contains any variables that aren't part of
+    `module`'s public API or that weren't declared with a `!default` flag in
+    `module`, throw an error.
```

Comment:
Oh right, of course - we're not waiting until execution to set the variables properly, that's already handled properly. We just wait until execution to throw any errors. Looks great! 

I have one more wording fix I want to cleanup